src/index.js

```diff
@@ -297,3 +304,127 @@ function canDisplayGraphiQL(
   // prefers HTML over JSON.
   return !raw && accepts(request).types([ 'json', 'html' ]) === 'html';
 }
+
+/**
+ * Helper function to resolve the Options to get OptionsData.
+ */
+function resolveOptionsData(
+  options: Options,
+  request: Request,
+  response: Response
+): Promise<OptionsData> {
+  return new Promise(resolve => {
```

Comment:
tried that, but doing so prevents us from catching exceptions thrown from options when it is a function at [1]